### DISCUSSION TITLE
CRUD: setuplist Row background color

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#525] [Q&A (Help)] [answered]
I have to highlight some rows based on a criterion, for example I would like to change the background color of the records that have the field "active" = false, do I have to use the [custom view](https://backpackforlaravel.com/docs/5.x/crud-how-to#how-to-customize-views-for-each-crud-panel-1) or is there a faster method?

Thanks

**Comment by blondie63** [comment]
@pxpm Hi Pedro, have you seen this ? :)

**Reply by pxpm** [reply]
Hey @blondie63 

Haven't had time to look at this. 

As far as I am aware there is nothing specific in Backpack that would help you to achieve that. 

About the implementation there are more than one path that lead to Rome, so either creating a custom view, or adding a JS widget hooked on the table draw event or something that adds some styling to some rows is up to you. 

I think you can get away without creating custom views, just using JS and the Datatables api to do your job.

Cheers

**Comment by arminkardovic** [comment, marked as answer]
Hi, 

I have a workaround if you are still interested,

add javascript into your global script file -- **app.js** and listen ajax complete function in jQuery

```javascript

$(document).ready(() => {
    $(document).ajaxComplete(function (event, jqXHR, ajaxOptions) {
        const project = new Project();
        project.checkRows();
    });
});
``` 

Make function that will check row, something like this
```javascript
class Project {
    checkRows() {
            $('[data-mark="true"]').each(function () {
                const $this = $(this);
                const min = $this.data('min');
                const current = $this.data('current');
    
                let elem = $this.closest("tr");
                if (min > current && current > 0) {
                    elem.css('background-color', '#ee8e8e');
                    elem.addClass("blinking");
                } else {
                    elem.css('background-color', 'initial');
                }
            });
        }
}
``` 

Add a new file, a custom column for the Laravel Backpack

> resources/views/vendor/backpack/crud/columns/row_id.blade.php

And add your custom code, in my case is like this
In Controller

```php

 CRUD::column('id')
            ->label("#ID")
            ->type("row_id");
``` 

```php
{{-- regular object attribute --}}
@php
    $column['value'] = $column['value'] ?? data_get($entry, $column['name']);
    $column['escaped'] = $column['escaped'] ?? true;
    $column['limit'] = $column['limit'] ?? 32;
    $column['prefix'] = $column['prefix'] ?? '';
    $column['suffix'] = $column['suffix'] ?? '';
    $column['text'] = $column['default'] ?? '-';

    if($column['value'] instanceof \Closure) {
        $column['value'] = $column['value']($entry);
    }

    if(is_array($column['value'])) {
        $column['value'] = json_encode($column['value']);
    }

    if(!empty($column['value'])) {
        $column['text'] = $column['prefix'].Str::limit($column['value'], $column['limit'], '…').$column['suffix'];
    }

    $current= $entry->finalMarkForProject ?? 0;
    $min = (float)config("settings.final_mark_for_project", 51);

@endphp

<span data-mark="true" data-min="{{$min}}" data-current="{{$current}}">
    @if($column['escaped'])
        {{ $column['text'] }}
    @else
        {!! $column['text'] !!}
    @endif
</span>

``` 
I hope that will help you,

**Comment by blondie63** [comment]
@arminkardovic Thanks for your share and sorry for delay..
